packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -545,6 +561,18 @@ export class GarbageCollector implements IGarbageCollector {
         );
     }
 
+    /**
+     * Called when a reference is added to a node. This is used to identify all nodes that have been referenced
+     * between summaries so that thier unreferenced timestamp can be reset. We will do that the next time GC runs;
+     * store the reference for now.
+     *
+     * @param fromNodeId - The node from which the reference is added.
+     * @param toNodeId - The node to which the reference is added.
+     */
+    public referenceAdded(fromNodeId: string, toNodeId: string) {
```

Comment:
One random though - instead of proxying calls through layers, consider passing some stripped-down GC interface down to DDSs. Some pros & cons of this approach:
- It's less proxying code, less requirements to data store runtime implementers (possibly).
- It's easier to make it optional (i.e. implementer of container runtime may decide not to bother and not implement GC, and DDSs can work just fine - same can be achieved today, but empty methods would need to be implemented).
- Con: such interface is on compat boundary, it's less obvious it can't easily be changed. People are more used to a fact that you can't easily change interop between data store and container runtime layers.

Reply:
That's a good suggestion. We will need 2 GC interfaces - one for communicating up the layers and one for communicating down the layers. Layers that participate in GC will need to implement these interfaces at the right places.
I think this needs a separate PR of its own as the changes will be big and might run into issues.

Reply:
Create an issue to track this - https://github.com/microsoft/FluidFramework/issues/8578